### DISCUSSION TITLE
Start "Play simulation" programmatically

### DISCUSSION THREAD

**Discussion by ppueyor** [microsoft/AirSim#4456] [Support Q&A]
Hi!

Does anyone know how to start/stop the simulation in the editor programmatically?

It would be useful to run unsupervised tests to extract a big amount of data.

Thanks!

**Comment by Jamy-L** [comment]
I don't think it is the best way of doing that, for different reasons. Thirst of all, it is better to launch the project than simply playing it, because then the simulation is not capped at 120FPS anymore, meaning you can reach higher clock speed without losing much physic accuracy, so you could run more simulation in a given time. Second, thing is that you would lose a valuable time when disconnecting and connecting again to the client. My best advise would be to launch the sim manually at the beginning, and use the client.reset command of the python API, along with simPause. You can also use the API to set inital speed, position etc.

The only flaw to that solution is that you cannot change the settings.JSON or the vehicle pawn blueprints, so you would have to stick with the same physical parameters until the end. If programmatically launching the project really is the only way, you should search in the Unreal Engine 4 API, there is probably a command to launch it.

**Reply by ppueyor** [reply]
Thanks for the answer!

Very useful info!
